FunctionalTableDataDemo/Cells/Button/ButtonState.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  ButtonState.swift
+//  Shopify
+//
+//  Created by Raul Riera on 2017-05-02.
+//  Copyright © 2017 Shopify. All rights reserved.
+//
+
+import UIKit
+import FunctionalTableData
+
+typealias ButtonCell = HostCell<UIButton, ButtonState, LayoutMarginsTableItemLayout>
+
+public struct ButtonState: Equatable {
+	public let title: String
+	public let isEnabled: Bool
+	public let alignment: UIControl.ContentHorizontalAlignment
+	public let action: (UIButton) -> Void
```

Comment:
Might be able to get away with a target/action approach instead of closure based.
A `nil` target with an action that exists on the controller that the view will be inside should, I think, fire correctly (and TBH is a nice demo of how the responder chain works/was intended to work).
Alternatively an actual target/action object/selector captured here could also be nicer (target held onto weakly, state comparison with == can then actually compare the target via `===` (identity) and the action directly).

Haven't tried either of these approaches, but my opinion is that they lead to better leveraging of the Cocoa APIs, and they also lead in a direction where the state instances can be made Codable (minus the target pointing to an instance, that's not going to work)

Reply:
Could also ignore this comment for now and try it after merging...

Reply:
Would you delete the backing UIControl extension as well? That will completely remove all the code Tom built 😏